contracts/collectxyz-nft-contract/src/query.rs

```diff
@@ -154,6 +162,49 @@ pub fn query_all_nft_info(
     })
 }
 
+pub fn query_tokens(
+    deps: Deps,
+    owner: String,
+    start_after: Option<String>,
+    limit: Option<u32>,
+) -> StdResult<Cw721TokensResponse> {
+    let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+    let start = start_after.map(Bound::exclusive);
+
+    let owner_addr = deps.api.addr_validate(&owner)?;
+    let pks: Vec<_> = tokens()
+        .idx
+        .owner
+        .prefix(owner_addr)
+        .keys(deps.storage, start, None, Order::Ascending)
+        .take(limit)
+        .collect();
+
+    let res: Result<Vec<_>, _> = pks.iter().map(|v| String::from_utf8(v.to_vec())).collect();
+    let tokens = res.map_err(StdError::invalid_utf8)?;
+    let numeric_tokens: Vec<String> = tokens.iter().map(|s| numeric_token_id(s.to_string())).collect();
+    Ok(Cw721TokensResponse { tokens: numeric_tokens })
+}
+
+pub fn query_all_tokens(
+    deps: Deps,
+    start_after: Option<String>,
+    limit: Option<u32>,
+) -> StdResult<Cw721TokensResponse> {
+    let limit = limit.unwrap_or(DEFAULT_LIMIT).min(MAX_LIMIT) as usize;
+    let start = start_after.map(Bound::exclusive);
+
+    let res: StdResult<Vec<String>> = tokens()
+        .range(deps.storage, start, None, Order::Ascending)
+        .take(limit)
+        .map(|item| item.map(|(k, _)| String::from_utf8_lossy(&k).to_string()))
+        .collect();
+        
+    let tokens = res.map_err(StdError::invalid_utf8)?;
+    let numeric_tokens: Vec<String> = tokens.iter().map(|s| numeric_token_id(s.to_string())).collect();
+    Ok(Cw721TokensResponse { tokens: numeric_tokens })
+}
+
```

Comment:
should we write a simple unit test for this to test saving tokens before calling this?